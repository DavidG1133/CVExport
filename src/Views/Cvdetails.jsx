import React, { useState, useEffect } from 'react'
import { useParams } from 'react-router-dom'
import CvTemplate from '../Components/CvTemplate'
import cvTemplates from '../ViewModels/cvTemplates'
import '../Views/Styles/cvdetails.css'

const Cvdetails = () => {

    const { id } = useParams();
    const [cvData, setCvData] = useState(null)

    useEffect(() => {
        const template = cvTemplates.find(t => t.id === parseInt(id));
        setCvData(template);
    }, [id])

    const handleChange = (e) => {
        const { name, value } = e.target;
        setCvData(prevState => ({
            ...prevState,
            [name]: value
        }));
    };

    if (!cvData) {
        return <div>Loading...</div>
    }

    return (
        <>
            <div className="cv-details-container">
                <div className="cv-form">
                    <h2>Edit Your CV</h2>
                    <label>
                        Name:
                        <input type="text" name="name" value={cvData.name} onChange={handleChange} />
                    </label>
                    <label>
                        Title:
                        <input type="text" name="title" value={cvData.title} onChange={handleChange} />
                    </label>
                    <label>
                        Summary:
                        <textarea name="summary" value={cvData.summary} onChange={handleChange} />
                    </label>

                </div>
                <CvTemplate {...cvData} />
            </div>
        </>
    )
}

export default Cvdetails