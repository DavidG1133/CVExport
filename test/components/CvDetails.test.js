import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Cvdetails from '../../src/Views/Cvdetails';

it('should generate PDF', () => {
  const { getByText } = render(
    <MemoryRouter>
      <Cvdetails />
    </MemoryRouter>
  );
  const generatePdfButton = getByText('Generar PDF');
  fireEvent.click(generatePdfButton);
  expect(window.jsPDF).toHaveBeenCalledTimes(1);
});

it('should match snapshot', () => {
  const { container } = render(
    <MemoryRouter>
      <Cvdetails />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});