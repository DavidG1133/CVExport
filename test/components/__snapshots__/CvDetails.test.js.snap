// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <nav
    class="navbar2 navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <header>
        <img
          alt="Logo"
          class="logo-image"
          src="[object Object]"
          style="position: absolute; top: 20px; left: 50px; width: auto; height: auto; max-width: 100%; max-height: 20%;"
        />
      </header>
      <div
        class="navbar-nav ml-auto"
      >
        <div
          class="icons"
        >
          <a
            class="nav-link"
            href="/"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-8zdB9ITEokTy"
              class="svg-inline--fa fa-arrow-right fa-icon"
              data-icon="arrow-right"
              data-prefix="fas"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-8zdB9ITEokTy"
              >
                Salir
              </title>
              <path
                d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="cv-details"
  >
    <h1>
      Detalles del CV
    </h1>
    <form>
      <div>
        <label>
          Foto:
        </label>
        <input
          name="foto"
          type="file"
        />
      </div>
      <div>
        <label>
          Nombre:
        </label>
        <input
          name="nombre"
          placeholder="Nombre/Apellidos"
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Experiencia Laboral:
        </label>
        <input
          name="experienciaLaboral"
          placeholder="Experiencia laboral"
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Estudios:
        </label>
        <input
          name="estudios"
          placeholder="Nivel de estudios (ingenieria, licenciatura, otros)"
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Habilidades y competencias profesionales:
        </label>
        <input
          name="habilidades"
          placeholder="Habilidades y competencias"
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Idiomas:
        </label>
        <input
          name="idiomas"
          placeholder="Idiomas"
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Sobre Mi:
        </label>
        <input
          name="sobreMi"
          placeholder="Describete"
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Herramientas de software
        </label>
        <input
          name="herramientas"
          placeholder="Lenguajes, frameworks, etc."
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          Contacto:
        </label>
        <input
          name="contacto"
          placeholder="Telefono"
          type="text"
          value=""
        />
        <input
          name="correo"
          placeholder="Correo electronico"
          type="text"
          value=""
        />
        <input
          name="direccion"
          placeholder="Domicilio"
          type="text"
          value=""
        />
      </div>
      <button>
        Generar PDF
      </button>
      <div>
        <button>
          <a
            href="/SubIndex"
          >
            Regresar
          </a>
        </button>
      </div>
    </form>
  </div>
</div>
`;
